src/batch/src/executor/table_function.rs

```diff
@@ -50,19 +52,25 @@ impl TableFunctionExecutor {
     async fn do_execute(self: Box<Self>) {
         let dummy_chunk = DataChunk::new_dummy(1);
 
-        let mut builder = self
-            .table_function
-            .return_type()
-            .create_array_builder(self.chunk_size);
-        let mut len = 0;
+        // let mut builder = self
+        //     .table_function
+        //     .return_type()
+        //     .create_array_builder(self.chunk_size);
+
+        let mut data_chunk_builder =
+            DataChunkBuilder::new(vec![self.table_function.return_type()], self.chunk_size);
         for array in self.table_function.eval(&dummy_chunk).await? {
-            len += array.len();
-            builder.append_array(&array);
+            let len = array.len();
+            for chunk in data_chunk_builder
+                .append_chunk(DataChunk::new(vec![Column::new(array)], Vis::from(len)))
```

Comment:
Perhaps the original match should be kept to handle the multiple return values case.
```suggestion
            let chunk = match array {
                ArrayImpl::Struct(s) => DataChunk::from(s),
                array => DataChunk::new(vec![array.into()], len),
            };
            for chunk in data_chunk_builder.append_chunk(chunk)
```